### ISSUE TITLE
StartMatchmaking does not work when matchmaking=false

### ISSUE THREAD

**Issue by SaadBazaz** [asadm/playroom-unity#58]
### What happened?

In Unity, if I call InsertCoin with matchmaking == true, if seems to work fine. 
But I call InsertCoin with matchmaking == false, then call StartMatchmaking after that, I get an error.

Here is my code:
```csharp
private IEnumerator StartMatchmakingCoroutine()
{
    NetworkManager.Instance.Initialize();       //  Call InsertCoin with matchmaking = false

    yield return new WaitUntil(() => NetworkManager.Instance.GetPlayers().Count > 0);

    NetworkManager.Instance.StartMatchmaking(); //  Call StartMatchmaking
}
```
\

### Version

0.0.14

### What is your environment?

Unknown

### Link to original discussion

https://discord.com/channels/997752993598419044/1228773906710528144

### Relevant log output

In the console log image:
1. the first line is printed after InsertCoin, in the OnPlayerJoin callback.
2. the second line is printed after StartMatchmaking, in the OnQuit callback
3. the third line (error line) is printed just after that

![image](https://github.com/asadm/playroom-unity/assets/51885228/7fdf3411-0301-41a1-90c9-3d7f1d5cbc6b)

After that, I get even more errors:

![image](https://github.com/asadm/playroom-unity/assets/51885228/3db7e828-d09e-455e-b240-0554597f04c3)

**Comment by SaadBazaz** [comment]
## Current recommendation

Instead of setting `matchmaking = false`, leave it `null` or `undefined`. 

For example, instead of:
```csharp
PlayroomKit.insertCoin(
  matchmaking=false, 
  //... other options
)
```

You could try:
```csharp
PlayroomKit.insertCoin(
  //... other options, no matchmaking option at all
)
```

## Next steps
Will close the Issue if the above recommendations yield positive results, or if the Issue become stale.

**Comment by SaadBazaz** [comment]
Paraphrasing @momintlh:

> I have tested the StartMatchMaking one, it first works fine then if a player leaves the game will crash and the attached ID error is thrown. after that I get the other abnormal behavior error, that is due to the coroutine, provided in the example.
![image](https://github.com/asadm/playroom-unity/assets/51885228/01b7f641-e729-4049-860a-b450c3692ac5)

> Not setting matchmaking within insertcoin's initoption leads to the same result because its default value is false. It cannot be null because it's a C# bool.

Marking this as an issue which needs to be solved.

**Comment by SaadBazaz** [comment]
> Going to test without using GetPlayers(), I will maintain a list on the games side and use that.

That sounds like a good idea to narrow it down. Did it work?

**Comment by momintlh** [comment]
> > Going to test without using GetPlayers(), I will maintain a list on the games side and use that.
> 
> That sounds like a good idea to narrow it down. Did it work?

Getting the same error

**Comment by SaadBazaz** [comment]
If you can share an alt snippet which people can use instead of using it incorrectly, that could work as a solution too.

**Comment by momintlh** [comment]
The code below should achieves the same goal and leads to the same error

```c#
private void Initialize()
    {
        PlayroomKit.InsertCoin(new PlayroomKit.InitOptions()
        {
            maxPlayersPerRoom = 2,
            roomCode = "abc",
            matchmaking = false,
            defaultPlayerStates = new() {
                        {"score", 0},
                    },

        }, () =>
        {
            PlayroomKit.OnPlayerJoin(AddPlayer);
            PlayroomKit.StartMatchmaking();
        });
    }
```

From my understanding the error is occurring when the room changes.

**Comment by momintlh** [comment]
The fix might be this: Wait for the room change, (maybe by reading the URL) and only then call MyPlayer / or Update. I'll try this and report.

**Comment by SaadBazaz** [comment]
Resolved in #70
